x/mm2/keeper/msg_server.go

```diff
@@ -142,6 +142,41 @@ func (ms msgServer) Params(goCtx context.Context, msg *types.MsgParams) (*types.
 	return &types.MsgParamsResponse{}, nil
 }
 
+func (ms msgServer) RemoveMarketAuthorities(goCtx context.Context, msg *types.MsgRemoveMarketAuthorities) (*types.MsgRemoveMarketAuthoritiesResponse, error) {
+	if msg == nil {
+		return nil, fmt.Errorf("unable to process nil msg")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params, err := ms.k.GetParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if msg.Admin != params.Admin {
+		return nil, fmt.Errorf("request admin %s does not match module admin %s", msg.Admin, params.Admin)
+	}
+
+	if len(msg.RemoveAddresses) > len(params.MarketAuthorities) {
+		return nil, fmt.Errorf("remove addresses must be a subset of the current market authorities")
+	}
+
+	for i, address := range params.MarketAuthorities {
+		for _, remove := range msg.RemoveAddresses {
```

Comment:
instead of iterating thru all addreses to remove, construct a mapping of them, and iterate thru the market-authorities and if included in addresses to remove, remove O(n^2) -> O(n)